Migrate HCA client code to vets-website.

awong-dev opened the pull request.

awong-dev committed:
Pull HCA code into temporary migration directory.

awong-dev committed:
Move source and most tests into right directories.

This does NOT build yet.

awong-dev committed:
Make HCA Build. Styles and tests still broken.

But this is much closer!

jbalboni commented on config/webpack.config.js:
```diff
@@ -80,7 +83,7 @@ var configGenerator = (options) => {
     plugins: [
       new webpack.DefinePlugin({
-          __BUILDTYPE__: options.buildtype
+          __BUILDTYPE__: JSON.stringify(options.buildtype)
```
Is `process.env.NODE_ENV` set to `"production"` in any of the Webpack configs? It seems like this is building React (and other libs) in dev mode all the time (and there's an error to that effect on the prod vets.gov site).

(I'm new, so apologies if this is taken care of elsewhere or already been talked about)

awong-dev replied:
Good catch!

Yes, that's been a long-standing bug.  It's actually set in travis now (wee .travis.yml) and should go away.

It didn't quite belong in the webpack.config.js because sometimes you do want to build the dev version.

Fingers crossed that it will work this time...

jbalboni replied:
I'm not super knowledgeable about Travis, but I think that will just set NODE_ENV for the Node process that's running Webpack? Webpack itself also needs to be configured, through the define plugin, to replace NODE_ENV in the bundle it's building. I think it would look something like:

```
new webpack.DefinePlugin({
      __BUILDTYPE__: JSON.stringify(options.buildtype),
     'process.env': {
         'NODE_ENV': JSON.stringify(process.env.NODE_ENV)
     }
})
```

awong-dev replied:
added.

awong-dev committed:
Use the right entry.

awong-dev committed:
Add back in hca SCSS.

awong-dev committed:
Dedup sytles between _shame and hca.

Also reorder the variable loading. Importing our uswds variables before
importing all the uswds templates seems to undo our settings for the
usa-max-width.

awong-dev committed:
Migrate one unit test.

crwallace committed:
import ha styles into bundle

awong-dev committed:
Use webkit to split the style file

awong-dev committed:
Migrate HCA unittests.

Most tests run now except for ones that expect a document. Those fail
and need to be split into an e2e test.

awong-dev committed:
Use jsdom to fix unittests.

awong-dev committed:
Disable test that breaks on JSDOM.

Unittests are all green now.

awong-dev committed:
Disable nightwatch for now.

awong-dev committed:
Remove line-length rule and unneeded globals.

HCA now lints clean.

awong-dev committed:
Clean up stylesheet and entrypoint loading.

Now the page templates handle an entrypoint per app along with a style
per app.

awong-dev committed:
Update shrinkwrap

awong-dev commented:
PR ready for review.

One Unittest was disabled and nightwatch still doesn't work but I expect to fix that in a follow-up PR.

awong-dev committed:
Add jsdom into package.json.

Left it out accidentally. Oops.

akainic commented on test/common/model/fields.spec.js:
```diff
@@ -0,0 +1,90 @@
+const chai = require('chai');
```
Do we not want to nest this file under an `hca` directory in the `test` directory? Same with a few of the other test files here.

awong-dev replied:
This was intentional. I was hoping/expecting that the models would be shared.

akainic commented:
Is the purpose of the `hca-migrate` directory just to store some files temporarily that came exactly from the healthcare-application repo?

akainic commented on src/js/hca/components/form-elements/ErrorableSelect.jsx:
```diff
@@ -0,0 +1,131 @@
+import React from 'react';
```
Maybe in a future PR we can move these form-element or question components to a directory outside of HCA so they can be used by other apps, but for now this is good :)

awong-dev replied:
Yes!  I reason I want to get this in so bad is because we need to enable all the current development to refactor up common components.

awong-dev commented:
The hca-migrate directory was the original landing spot for the hca code. The code that's left there is stuff i'm still trying to move.  It's mostly used by e2e tests I think which I don't quite have working yet.  But that directory should go away.

awong-dev committed:
Delete propType tests.

These were fragile and overkill. Also, when the build is set to
production, proptype checks are all removed from the react codebase
causing all these tests to fail. Since they were not providing much
value and they were confusing and fragile, it's better to just remove
them.

awong-dev committed:
Merge branch 'master' into hca

akainic commented:
Tested this locally, specifically for HCA. Everything worked well except for actually submitting the form at the end, which seems to work locally in the healthcare-application repo, so not sure what's going on there.

awong-dev commented:
That's sorta broken as expected for now

On Tue, Aug 16, 2016, 10:39 Anne Kainic <notifications@github.com> wrote:

> Tested this locally, specifically for HCA. Everything worked well except
> for actually submitting the form at the end, which seems to work locally in
> the healthcare-application repo, so not sure what's going on there.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/department-of-veterans-affairs/vets-website/pull/2703#issuecomment-240122295>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAI0w3_ogaW6rEYNmKpzxQ9e2AH8dq0Xks5qgct2gaJpZM4JglHa>
> .
>

awong-dev committed:
Merge branch 'master' into hca
